clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -100,11 +100,10 @@ public RuntimeException exception() {
      * @param value corresponding value (or null if there is none)
      */
     public void complete(T value) {
-        if (isDone)
+        if (!isDone.compareAndSet(false, true))
```

Comment:
Btw, can `complete` and `raise` methods actually be called from multiple threads concurrently? Or are we just trying to be safe in case that happens in the future? I understand that the accessors methods can be called from multiple threads.